src/FsCheck/Arbitrary.fs

```diff
@@ -846,9 +846,14 @@ module Arb =
 
         ///Generates an interval between two non-negative integers.
         static member Interval() =
-            gen { let! start,offset = Gen.two generate
-                  return Interval (abs start,abs start+abs offset) }//TODO: shrinker
-            |> fromGen
+            let generator = 
+                  gen { let! start,offset = Gen.two generate
+                  return Interval (abs start,abs start+abs offset) }
+            let shrinker (i : Interval) =
+                (i.Left, i.Right - i.Left)
+                |> shrink
+                |> Seq.map (fun (right, length) -> Interval(right, right + length))
```

Comment:
nit: this should be `fun (left, length) ->` etc

Reply:
and actually, while we're at it, might as well make the terminology consistent between generator and shrinker. Either `left` `right` and `length` or `start` and `offset`. Would prefer the former.